registry/store/redis_registry.go

```diff
@@ -42,50 +42,56 @@ func NewRedisRegistry(db database.Database) ExternalRegistry {
 }
 
 func (rr *redisRegistry) ReadKeys(namespace auth.Namespace) ([]string, error) {
-	return rr.db.ReadKeys(namespace.String())
+	return rr.db.ReadKeys(fmt.Sprintf("%s%s%s", "*", keySeparator, namespace.String()))
 }
 
 func (rr *redisRegistry) ReadServiceInstanceByInstID(namespace auth.Namespace, instanceID string) (*ServiceInstance, error) {
-	siKey := fmt.Sprintf("%s.*", instanceID)
+	siKey := fmt.Sprintf("%s%s%s", instanceID, keySeparator, namespace.String())
 
-	keys, err := rr.ListServiceInstancesByKey(namespace, siKey)
+	entry, err := rr.db.ReadEntry(siKey)
 
 	if err != nil {
 		return nil, err
 	}
 	// If a key is not found, return an empty instance
-	if len(keys) == 0 {
+	if len(entry) == 0 {
 		var emptyServiceInstance ServiceInstance
 		return &emptyServiceInstance, nil
```

Comment:
len(entry) == 0 is a valid response (registering a new instance for example).  I returned an empty instance because we check the instance.ID during registration.  I can instead return nil and check instance != nil.